Makefile

```diff
@@ -352,7 +352,23 @@ lint-config-check:
 	$(GOTOOL) $(GOLINT_PKG) config verify -v
 
 #? lint: Run static code analysis
-lint: check-go-version lint-config-check lint-source 
+lint: check-go-version lint-config-check lint-source
+
+# Globs to exclude generated files from ast-grep.
+AST_GREP_EXCLUDE := --globs '!**/*.pb.go' --globs '!**/*.pb.gw.go' --globs '!**/*.pb.json.go' --globs '!**/sqlc/*.go'
+
+# Optional directory/package filter for ast-grep (e.g., make ast-lint pkg=lnwallet).
+AST_GREP_PATH := $(if $(pkg),$(pkg),.)
+
+#? ast-lint: Run ast-grep style checks. Use pkg=<dir> to focus on a specific directory.
+ast-lint:
```

Comment:
Looks like it giving false positives, or should we update our guidelines to be more specific?
```
warning[func-def-dangling-return-paren]: Function return types should not start with open paren on its own line
     ┌─ lnwallet/channel_test.go:1625:1
     │  
1625 │ ╭ func TestHTLCSigNumber(t *testing.T) {
     · │
1775 │ │ }
     │ ╰─^
     │  
     = Per docs/development_guidelines.md: Return type wrapping should continue
       on the same line, not start a new line with open paren.
       
       WRONG:
         func bar(a, b, c) (
         d, error,
         ) {
       
       RIGHT:
         func baz(a, b, c) (d,
         error) {
```

Reply:
Yeah it certainly needs some tweaking. `ast-grep` is based on an AST so it makes certain types of replace+search easier, but the items that we want to restrict here include whitespace, etc -- which doesn't neccesarily show up in the AST.